### ISSUE TITLE
Doc planning for 5.0

### ISSUE THREAD

**Issue by jdfreder** [ipython/ipywidgets#306]
Meta issue to discuss what needs to be done for the widget 5.0 documentation.

cc @ellisonbg @SylvainCorlay @willingc @captainsafia @lbustelo

**Comment by captainsafia** [comment]
Here's what I came up with from a cursory exploration of the current documentation.

- [ ] Update command used for executing JavaScript tests
- [ ] Convert the Widgets List.ipynb notebook into RST and include in documentation under examples section
- [ ] Transfer interact documentation notebook into an RST file

I think it'll be a good idea to consolidate all of the important documentation in one place. Hence, the documentation that was previously in notebooks should be fully or partially reproduced in the Sphinx docs. What does everyone else think?

**Comment by jdfreder** [comment]
I think a fundamental problem with the widget docs is that the notebook front-end is required to render the widgets themselves.  We have a lot of good tutorial-ish content, which when converted to MD or RST becomes hard to follow because you can't see the output of the examples.

This leads me to believe we need a way to generate screenshots or static representations of the widgets that are compatible with RTD.  I'll investigate this today.

> Convert the Widgets List.ipynb notebook into RST and include in documentation under examples section

Actually, I just talked to Brian about this, and he had a good idea.  

(background) Ever since I striped out the casper.js tests, the widgets don't have any round trip tests.  Casper.js was a good tool (maybe I'm giving it too much credit here) for conducting high level GUI tests.  With the new Karma setup, I hope to have roundtrip unit tests, but I don't plan on adding GUI tests.  

(Brian's idea) Instead, we should get rid of the widget list notebook altogether, and create a notebook for each widget, illustrating what we claim that should work for that widget.  Then, prior to a release, we will review the notebooks by hand to verify all is well.  These notebooks become QA notebooks.

> I think it'll be a good idea to consolidate all of the important documentation in one place. Hence, the documentation that was previously in notebooks should be fully or partially reproduced in the Sphinx docs. What does everyone else think?

I **definitely agree** that they should be rendered in one place, namely RTD.  However, I don't think we can get away from having some notebooks, just because of the subject matter (widgets).  Either we do the opposite, and convert the RST content all to notebooks or we live with a mix of the two.

This brings me back to my investigation of generating widget screenshots or static representations that will work on RTD.  

I'll keep everyone posted by commenting back here.

**Comment by willingc** [comment]
@jdfreder Thanks for kicking off the planning and looping me in. A few thoughts...

**Documenting widget interactivity**  Sphinx will likely never measure up to be a fabulous option for demonstrating dynamic content, such as an interactive widget. I really like the idea of test widgets notebooks that then become released notebooks. It's better for the user and they would be easy enough to link to from the Sphinx documentation.

Static info and GIFs in Sphinx - Likely the best option for a visual illustration of the individual widget would be:
- brief text description of the widget (in a RST file)
- a static screenshot or thumbnail of the widget (in a RST file)
- a link to the interactive widget test notebook (rendered outside of Sphinx) [Note: the test notebook would be the definitive resource for how the widget works]
- and, possibly, link to a GIF of the interactive widget (stored in Sphinx docs source rendered with Sphinx and RTD) *This likely may not be necessary since the notebooks can cover this task for an individual widget; this may be useful for collections of widgets within a theme*

**Comment by SylvainCorlay** [comment]
One of the main goals for ipywidgets 6.0 should be to generate Github pages including live examples of widgets embedded like this one [bqplot example](http://www.proba.jussieu.fr/pageperso/corlay/bqplot.html) including a text area and a live (but backend-less) rendered version of the widget.

It would be fantastic if we could generate the same thing on RTD, but replaced with images.

**Comment by willingc** [comment]
@jdfreder @SylvainCorlay @captainsafia   Can this be closed? Or should the milestone be bumped to 6.0?

**Comment by SylvainCorlay** [comment]
We are pretty much getting there.

**Comment by jdfreder** [comment]
We still need to write some new content with the static widget stuff we got working.

**Comment by willingc** [comment]
Sounds good to me.

**Comment by willingc** [comment]
@SylvainCorlay Do you want to close this issue and open a new one for interactive documentation for 6.0 or just change edit the milestone and title of this issue?

**Comment by SylvainCorlay** [comment]
@willingc sure thing!
